simplecpp.cpp

```diff
@@ -377,6 +377,42 @@ class StdIStream : public simplecpp::TokenList::Stream {
     std::istream &istr;
 };
 
+class StdCharBufStream : public simplecpp::TokenList::Stream {
+public:
+    // cppcheck-suppress uninitDerivedMemberVar - we call Stream::init() to initialize the private members
+    StdCharBufStream(const unsigned char* str, std::size_t size)
+        : str(str)
+        , size(size)
+        , pos(0)
+        , lastStatus(0)
+    {
+        init();
+    }
+
+    virtual int get() OVERRIDE {
+        if (pos >= size)
+            return lastStatus = EOF;
+        return str[pos++];
+    }
+    virtual int peek() OVERRIDE {
+        if (pos >= size)
+            return lastStatus = EOF;
+        return str[pos];
+    }
+    virtual void unget() OVERRIDE {
+        --pos;
+    }
+    virtual bool good() OVERRIDE {
+        return lastStatus != EOF;
+    }
+
+private:
+    const unsigned char *str;
```

Comment:
const unsigned char* const str; ?

Reply:
Possibly - will check.

If we want to use that we could enable it in `misc-const-correctness` so it will be used consistently forced across the code.

Reply:
then.. feel free to use misc-const-correctness for members. but it's not necessary in this PR.

Reply:
We already have this enabled so it looks like a false negative.

Reply:
The check only checks local variables. I will still file a ticket upstream.

Reply:
I forgot to apply this change. Also suggested this upstream: https://github.com/llvm/llvm-project/issues/93282.